### PR TITLE
Re-introduce variables testing

### PR DESCRIPTION
**Pull request by bryanculver**
# Closes: #DNE
# What's Changed
- Reintroduce the jobs variables testing
- Rename what was Script to Job

# TODO

- [x] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/tests/test_job_variables.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        # Validate valid choice
+        data = {"var1": "ff0000"}
+        form = ChoiceVarJob().as_form(data)
+        self.assertTrue(form.is_valid())
+        self.assertEqual(form.cleaned_data["var1"], "ff0000")
+
+        # Validate invalid choice
+        data = {"var1": "taupe"}
+        form = ChoiceVarJob().as_form(data)
+        self.assertFalse(form.is_valid())
```
Comment (glennmatthews) [inline nautobot/extras/tests/test_job_variables.py:99]: Perhaps also add?

```python
self.assertIn("var1", form.errors)
```

To match other tests.

File: nautobot/extras/tests/test_job_variables.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        # Validate multiple choices
+        data = {"var1": ("ff0000", "00ff00")}
+        form = MultiChoiceVarJob().as_form(data)
+        self.assertTrue(form.is_valid())
+        self.assertEqual(form.cleaned_data["var1"], ["ff0000", "00ff00"])
+
+        # Validate invalid choice
+        data = {"var1": "taupe"}
+        form = MultiChoiceVarJob().as_form(data)
+        self.assertFalse(form.is_valid())
```
Comment (glennmatthews) [inline nautobot/extras/tests/test_job_variables.py:117]: Same suggestion here

File: nautobot/extras/tests/test_job_variables.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+    def test_objectvar(self):
+        # Populate some objects
+        for i in range(1, 6):
+            Role(name=f"Device Role {i}").save()
```
Comment (glennmatthews) [inline nautobot/extras/tests/test_job_variables.py:122]: Shouldn't be needed, I think? Don't we have factories for Role?

File: nautobot/extras/tests/test_job_variables.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+        # Validate valid data
+        data = {"var1": Role.objects.get_for_model(Device).first().pk}
+        form = ObjectVarJob().as_form(data)
+        self.assertTrue(form.is_valid())
+        self.assertEqual(form.cleaned_data["var1"].pk, data["var1"])
+
+    def test_multiobjectvar(self):
+        # Populate some objects
+        for i in range(1, 6):
+            Role(name=f"Device Role {i}").save()
```
Comment (glennmatthews) [inline nautobot/extras/tests/test_job_variables.py:133]: Same comment

**Review by glennmatthews — APPROVED** [review]

**Review by HanlinMiao — APPROVED** [review]

**Pull request merged**